Civi.php

```diff
@@ -84,8 +84,8 @@ public static function lockManager() {
   /**
    * @return \CRM_Core_Error_Log
    */
-  public static function log() {
-    return Civi\Core\Container::singleton()->get('psr_log');
+  public static function log($log = 'psr_log') {
+    return Civi\Core\Container::singleton()->get($log);
```

Comment:
@totten - I realise thinking about this it would be good if we could fall back to the 'psr_log' if the service didn't exist - e.g say I uninstalled the extension that added a log but had sloppily used that log service from another extension it would be good for it to still load psr_log

Reply:
Hmm fallback would also prevent accidental bypassing of attempts to alter the logger https://lab.civicrm.org/extensions/monolog/-/blob/master/monolog.php#L181

Reply:
Yeah, I think if we're expanding `Civi::log()` to `Civi::log(string? $name)`, then it is important to provide a fallback mechanism.

Agree that this symbolic log name maps to Monolog "channels".

There is a part of me which would really prefer that we simply bring the Monolog dependency into `civicrm-core` so that we can take explicit advantage of its filters+channels. OTOH, there's always dep-hell (e.g. what happens if a Drupal site installs a monolog add-on  - or if a WP plugin incorporates Monolog). This PR avoids that by sticking to PSR interfaces and in-house implementations.

I've pushed  up a suggestion 14689ea which still keeps us strictly independent of Monolog - but threads-the-needle differently:

* The included `LogManager` is very simple. It does have a fallback, and it allows specific channels to be customized.
* An extension like https://lab.civicrm.org/extensions/monolog/ could replace the `psr_log_manager` -- and then use the `$channel` name in a Monolog-attuned way.

Of course, if you and @demeritcowboy were keen to fully assimilate to Monolog, then I'd also be OK with that. (Tho TBH... we'd probably still want something like `psr_log_manager`.)

Reply:
For the record -- the reason I think it's important to have a fallback mechanism is because `Civi::XXX()` functions are APIs available to extensions. Its more robust from an extension-writing POV if you can rely on `Civi::log('mything')` always working. 

However, providing a fallback does create a flip-side... typos/miscommunications will be silently ignored, eg

* `Civi::log('configuration')` vs `Civi::log('config')` ==> Two different logs, no complaints
* `Civi::log('ipn')` vs `Civi::log('IPN')` ==> Two different logs (on Linux) or one log (on MacOS/Windows), no complaints either way

I couldn't argue if someone said we should have a list of validated names. But I'm also happy to KISS. Maybe we should just list some recommended names in the docblock for `Civi::log()`.

Reply:
@totten I really like that - I'm wondering if we should limit the change at this stage to your commit as the taxonomy might warrant discussion whereas I think the commit that you added should be pretty straight forward to merge

Reply:
I'm kind of staying out of it because I didn't get back to that "extension" for a couple reasons. And I don't really have any bandwidth for more comments on the topic right now.

Reply:
@demeritcowboy no worries - with this merged I'm gonna use my new privelleges with the monolog extension to build a weight-loss empire and to play around with configuration options (for monolog - the former is just the commit method you suggested)

Reply:
li`PR`suction

Reply:
very good!

Reply:
Sharing the readme  for anyone confused

![image](https://user-images.githubusercontent.com/336308/114967243-3123a700-9ec8-11eb-9772-251bd4583ccc.png)